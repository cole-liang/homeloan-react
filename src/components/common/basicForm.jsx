import React, { Component } from "react";

import { Form, Button, InputGroup } from "react-bootstrap";

import styled from "styled-components";

const InputDiv = styled.div`
  & .inputToolTip {
    font-size: 13px;
    margin-bottom: 10px;
    color: rgba(0, 0, 0, 0.5);
  }

  & [id$="Icon"] {
    margin-left: 0px !important;
  }

  & .asterisk::before {
    content: "*";
    color: red;
    margin-right: 5px;
  }

  /* sm-md(small to middle screen)*/
  @media only screen and (max-width: 767.5px) {
    & .periodSelect select {
      width: 90px;
      padding: 3px;
    }
  }
`;

class BasicForm extends Component {
  renderInput = (
    path,
    values,
    touched,
    errors,
    handleChange,
    handleBlur,
    options = {}
  ) => {
    const {
      label,
      handleFocus,
      type = "text",
      posFeedback = "Looks good!",
      prepend,
      pwdStrengthMeter,
      placeholder,
      toolTip,
      passwordChange,
      required = false,
      periodSelect = null
    } = options;
    return (
      <InputDiv>
        <Form.Group controlId={`${path}Validation`}>
          {label && (
            <Form.Label>
              <div className="labelText">
                {required && <span className="asterisk" />}
                {label}
              </div>
            </Form.Label>
          )}
          {toolTip && (
            <div className="inputToolTip d-block d-md-none">{toolTip}</div>
          )}
          {passwordChange && pwdStrengthMeter}
          <InputGroup>
            {prepend && (
              <InputGroup.Prepend>
                <InputGroup.Text id={`${path}Prepend`}>
                  {prepend}
                </InputGroup.Text>
              </InputGroup.Prepend>
            )}
            <Form.Control
              type={type}
              name={path}
              value={values[path]}
              placeholder={placeholder ? placeholder : label}
              onChange={handleChange}
              onFocus={handleFocus}
              onBlur={handleBlur}
              isValid={posFeedback && touched[path] && !errors[path]}
              isInvalid={touched[path] && !!errors[path]}
            />
            <div className="periodSelect ml-2">{periodSelect}</div>
            {posFeedback && (
              <Form.Control.Feedback>{posFeedback}</Form.Control.Feedback>
            )}
            <Form.Control.Feedback type="invalid">
              {errors[path]}
            </Form.Control.Feedback>
          </InputGroup>
        </Form.Group>
      </InputDiv>
    );
  };

  renderSelect = (path, handleChange, options) => (
    <Form.Control as="select" name={path} onChange={handleChange}>
      {options.map(option => (
        <option key={option} value={option}>
          {option}
        </option>
      ))}
    </Form.Control>
  );

  renderCheckBox = (
    label,
    path,
    touched,
    errors,
    handleChange,
    options = {}
  ) => {
    const { required = false } = options;
    return (
      <InputDiv>
        <Form.Group>
          <Form.Check
            name={path}
            label={
              <div>
                {required && <span className="asterisk" />}
                {label}
              </div>
            }
            onChange={handleChange}
            isInvalid={touched[path] && !!errors[path]}
            feedback={errors[path]}
            id={`${path}Validation`}
          />
        </Form.Group>
      </InputDiv>
    );
  };

  renderSubmitButton = label => (
    <Button variant="outline-primary" type="submit" block>
      {label}
    </Button>
  );

  renderCustomButton = (
    label,
    onClick,
    variant = "outline-primary",
    type = "button"
  ) => (
    <Button variant={variant} type={type} onClick={onClick} block>
      {label}
    </Button>
  );

  renderCustomSubmitButton = (label, variant = "outline-primary") => (
    <Button variant={variant} type="submit" block>
      {label}
    </Button>
  );
}

export default BasicForm;
