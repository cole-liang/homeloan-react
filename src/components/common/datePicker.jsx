import React, { Component } from "react";
import DatePickerDialog from "material-ui/DatePicker/DatePickerDialog";
import { Button } from "react-bootstrap";
import { MuiThemeProvider } from "material-ui/styles";
import getMuiTheme from "material-ui/styles/getMuiTheme";
import { purpleA700 } from "material-ui/styles/colors";
import moment from "moment";
import styled from "styled-components";

const DatePickerDiv = styled.div`
  & i {
    size: 30px;
  }
  & button {
    position: relative;
    top: 32px;
  }
`;

let datePickerDialog;

/********** Need to find out how to change the color **********/
const materialTheme = getMuiTheme({
  overrides: {
    MuiPickersToolbar: {
      toolbar: {
        backgroundColor: purpleA700
      }
    }
  }
});

class DatePicker extends Component {
  openDatePicker() {
    datePickerDialog.show();
  }

  openDatePickerOnEnter(e) {
    if (e.key === "Enter") {
      this.openDatePicker();
    }
  }

  setDateReceived(date) {
    this.props.onChange(moment(date).format("DD/MM/YYYY"));
  }

  render() {
    const { initialDate, disableFuture } = this.props;
    const maxDate = disableFuture ? new Date() : null;
    return (
      <React.Fragment>
        <DatePickerDiv>
          <Button
            variant="light"
            onKeyUp={e => this.openDatePickerOnEnter(e)}
            onClick={() => this.openDatePicker()}
          >
            <i className="fa fa-calendar" />
          </Button>
        </DatePickerDiv>
        <MuiThemeProvider muiTheme={materialTheme}>
          <DatePickerDialog
            maxDate={maxDate}
            ref={r => (datePickerDialog = r)}
            /* Must provide firstDayOfWeek or rendering of calendar will be broken. */
            firstDayOfWeek={0}
            /* Don't close the dialog until the user presses 'Okay'. */
            autoOk={false}
            onAccept={date => this.setDateReceived(date)}
            initialDate={moment(initialDate, "DD-MM-YYYY").toDate()}
          />
        </MuiThemeProvider>
      </React.Fragment>
    );
  }
}

export default DatePicker;
