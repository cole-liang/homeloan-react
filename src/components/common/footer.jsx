import React from "react";
import styled from "styled-components";

import * as global from "../globalValues";

import { Row, Col } from "react-bootstrap";
import { Link } from "react-router-dom";

const FooterStyle = styled.footer`
  color: #9fa3a6;
  border-top: 1px solid #e2dee6;
  text-align: center;
  padding: 15px 0;
  background: #272727;

  & .companyBrand {
    opacity: 0.7;
    height: 40px;
    width: 40px;
    margin-top: -6px;
  }

  & .footerAbove {
    margin: 0px 20px;
    border-bottom: 1px solid #6c7073;
  }

  & .footerBelow {
    color: #6c7073;
    margin: 10px 20px 0px 20px;
  }

  & .footerAbove *,
  & .footerBelow * {
    display: inline-block;
  }

  & .footerAbout,
  & .footerBelow {
    font-size: 12px;
  }

  & .footerAbove li:hover,
  & .companyBrand:hover,
  & .legalList li:hover {
    color: #fff;
    opacity: 1;
    cursor: pointer;
  }

  & .footerNav {
    padding-left: 0px;
  }

  & .footerAbove ul {
    list-style: none;
    margin-bottom: 0px;
  }

  & .footerAbove li {
    padding: 0px 12px;
    height: 24px;
  }

  & .footerBelow li {
    padding-left: 9px;
  }

  && .footerSocial {
    margin-bottom: 7px;
  }

  & .dot {
    width: 2px;
    height: 2px;
    border: 1px solid #6c7073;
    display: inline-block;
    vertical-align: middle;
    margin-left: 8px;
    border-radius: 50%;
    background: #6c7073;
  }
`;

const Footer = () => {
  return (
    <FooterStyle>
      <Row className="footerAbove d-flex justify-content-center justify-content-md-between">
        <Col
          xs={12}
          md={6}
          className="footerAbout order-1 order-md-0 d-flex align-items-center justify-content-md-start justify-content-center noPaddingAndMargin"
        >
          <Link to="/">
            <img className="companyBrand" src={global.COMPANY_LOGO_URL} />
          </Link>
          <ul className="footerNav">
            <li>Who We Are</li>
            <li>Contact Us</li>
            <li>Careers</li>
          </ul>
        </Col>
        <Col
          xs={12}
          md={2}
          className="d-flex order-0 order-md-1 justify-content-center justify-content-md-end noPaddingAndMargin"
        >
          <ul className="footerSocial d-flex align-items-center noPaddingAndMargin">
            <li>
              <i className="fa fa-twitter-square" />
            </li>
            <li>
              <i className="fa fa-facebook-square" />
            </li>
            <li>
              <i className="fa fa-google-plus-square" />
            </li>
            <li>
              <i className="fa fa-instagram" />
            </li>
          </ul>
        </Col>
      </Row>
      <Row className="footerBelow d-flex justify-content-between">
        <Col
          xs={12}
          md={6}
          className="d-flex justify-content-center justify-content-md-start noPaddingAndMargin"
        >
          <span>
            Copyright <i className="fa fa-copyright" /> 2019{" "}
            {global.COMPANY_NAME} All Rights Reserved.
          </span>
        </Col>
        <Col
          xs={12}
          md={6}
          className="d-flex justify-content-center justify-content-md-end noPaddingAndMargin"
        >
          <ul className="legalList noPaddingAndMargin">
            <li>
              Privacy Policy
              <span className="dot" />
            </li>
            <li>
              Terms of Use
              <span className="dot" />
            </li>
            <li>Site Map</li>
          </ul>
        </Col>
      </Row>
    </FooterStyle>
  );
};

export default Footer;
