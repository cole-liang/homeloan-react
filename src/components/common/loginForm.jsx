import React from "react";
import BasicForm from "./basicForm";
import styled from "styled-components";

import { Formik } from "formik";
import { Form, Row, Col } from "react-bootstrap";

import * as yup from "yup";

const LoginFormDiv = styled.div`
  background: #f5f6f7;
  width: 100%;

  & #loginText {
    font-size: 20px;
  }
`;

const inputWidth = 6;

const usernameValid = "Please enter a valid email";
const usernameRequired = "Username is required";

const passwordRequired = "Password is required";

const schema = yup.object({
  username: yup
    .string()
    .email(usernameValid)
    .required(usernameRequired),
  password: yup.string().required(passwordRequired)
});

class LoginForm extends BasicForm {
  state = {
    initialValues: {
      username: "",
      password: ""
    }
  };

  render() {
    const { initialValues } = this.state;
    return (
      <Row noGutters>
        <Col md={7} />
        <Col>
          <LoginFormDiv>
            <Row id="loginText" className="justify-content-center">
              <span>LOG IN</span>
            </Row>
            <div className="inputFieids">
              <Formik
                validationSchema={schema}
                onSubmit={console.log}
                initialValues={initialValues}
              >
                {({
                  handleSubmit,
                  handleChange,
                  handleBlur,
                  values,
                  touched,
                  isValid,
                  errors
                }) => {
                  const commonAttrs = [
                    values,
                    touched,
                    errors,
                    handleChange,
                    handleBlur
                  ];
                  return (
                    <Form noValidate onSubmit={handleSubmit}>
                      <Row noGutters className="justify-content-center">
                        <Col xs={inputWidth}>
                          {this.renderInput("username", ...commonAttrs, {
                            prepend: <i className="fa fa-user" />,
                            posFeedback: null,
                            placeholder: "Username"
                          })}
                        </Col>
                      </Row>

                      <Row noGutters className="justify-content-center">
                        <Col xs={inputWidth}>
                          {this.renderInput("password", ...commonAttrs, {
                            text: "password",
                            prepend: <i className="fa fa-lock" />,
                            posFeedback: null,
                            placeholder: "Password"
                          })}
                        </Col>
                      </Row>
                      <Row noGutters className="justify-content-center">
                        <Col xs={inputWidth}>{this.renderButton("Login")}</Col>
                      </Row>
                    </Form>
                  );
                }}
              </Formik>
            </div>
          </LoginFormDiv>
        </Col>
      </Row>
    );
  }
}

export default LoginForm;
