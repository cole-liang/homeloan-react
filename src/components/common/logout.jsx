import { Component } from "react";
import { connect } from "react-redux";

import * as userAction from "../../actions/userAction";

class Logout extends Component {
  componentDidMount() {
    this.props.logout();
    this.props.history.push("/");
  }

  render() {
    return null;
  }
}

const mapDispatchToProps = dispatch => ({
  logout: () => dispatch(userAction.logoutUser())
});

export default connect(
  null,
  mapDispatchToProps
)(Logout);
