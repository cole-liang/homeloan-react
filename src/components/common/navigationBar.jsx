import React from "react";
import { Nav, Navbar, NavDropdown } from "react-bootstrap";
import { LinkContainer } from "react-router-bootstrap";
import { connect } from "react-redux";

import styled from "styled-components";
import * as global from "../globalValues";

const NavDiv = styled.div`
  & img {
    height: 40px;
    width: 40px;
    margin-top: -10px;
    margin-bottom: -10px;
  }

  & .avatar {
    display: inline-block;
    margin-right: 8px;
    height: 35px;
    width: 35px;
    border-radius: 100%;
    background-size: cover;
    background-image: url("https://ya-webdesign.com/images/avatar-icon-png-17.png");
  }

  & .navDropdown a.dropdown-toggle {
    display: flex;
    align-items: center;
    padding: 0px;
  }

  @media only screen and (min-width: 768px) {
    & .dropdown-menu {
      left: -74px;
    }
  }
`;

const NavigationBar = ({ user }) => {
  return (
    <NavDiv>
      <Navbar collapseOnSelect expand="md" bg="dark" variant="dark">
        <LinkContainer to="/">
          <Navbar.Brand>
            <img src={global.COMPANY_LOGO_URL} alt="Zinq Logo" />
            {global.COMPANY_NAME}
          </Navbar.Brand>
        </LinkContainer>
        <Navbar.Toggle aria-controls="responsive-navbar-nav" />
        <Navbar.Collapse id="responsive-navbar-nav">
          <Nav className="mr-auto">
            <LinkContainer to="/getStarted">
              <Nav.Link>Get Started</Nav.Link>
            </LinkContainer>
          </Nav>
          <Nav>
            {(!user || !!user.error) && (
              <React.Fragment>
                <LinkContainer to="/register">
                  <Nav.Link>Register</Nav.Link>
                </LinkContainer>
                <LinkContainer to="/login">
                  <Nav.Link>Login</Nav.Link>
                </LinkContainer>
              </React.Fragment>
            )}
            {!!user && !user.error && (
              <div className="navDropdown mr-md-2">
                <NavDropdown
                  title={
                    <React.Fragment>
                      <span className="d-none d-md-flex align-items-center">
                        <div className="avatar" />
                        <span>{user.firstName}</span>
                      </span>
                      <span className="d-block d-md-none">
                        <span>{`${user.firstName} ${user.lastName}`}</span>
                      </span>
                    </React.Fragment>
                  }
                  id="basic-nav-dropdown"
                >
                  <NavDropdown.Item href="/logout">Logout</NavDropdown.Item>
                </NavDropdown>
              </div>
            )}
          </Nav>
        </Navbar.Collapse>
      </Navbar>
    </NavDiv>
  );
};

const mapStateToProps = state => {
  return {
    user: state.user
  };
};

export default connect(
  mapStateToProps,
  null
)(NavigationBar);
