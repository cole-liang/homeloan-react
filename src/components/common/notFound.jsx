import React from "react";

import { Row } from "react-bootstrap";

import styled from "styled-components";

const NotFoundDiv = styled.div`
  padding: 100px 0px;
  height: 100%;
  display: flex;

  & .notFoundBg {
    background: #fff;
  }

  & .notFoundText,
  & .notFoundDesc {
    width: 100%;
    text-align: center;
  }

  & .notFoundText {
    font-size: 300px;
    font-weight: 600;
    color: rebeccapurple;
  }

  & .notFoundDesc {
    color: rgba(0, 0, 0, 0.7);
    font-size: 25px;
  }

  /* display: md(middle screen) */
  @media only screen and (max-width: 767.5px) {
    & .notFoundText {
      font-size: 150px;
    }

    & .notFoundDesc {
      font-size: 20px;
    }
  }
`;

const NotFound = () => {
  return (
    <NotFoundDiv>
      <Row
        noGutters
        className="noPadding notFoundBg flex-grow-1 justify-content-center align-items-center"
      >
        <div>
          <div className="notFoundText">404</div>
          <div className="notFoundDesc mb-5 px-2">
            Oops, your page is magically gone, please try again.
          </div>
        </div>
      </Row>
    </NotFoundDiv>
  );
};

export default NotFound;
