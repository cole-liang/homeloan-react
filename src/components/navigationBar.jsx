import React from "react";
import { Nav, Navbar, NavDropdown } from "react-bootstrap";
import { LinkContainer } from "react-router-bootstrap";

const companyLogoPath =
  "http://www.zinqsolutions.com.au/wp-content/uploads/2018/12/zinq-comingsoon-logo2.png";

const NavigationBar = () => {
  return (
    <Navbar collapseOnSelect expand="lg" bg="dark" variant="dark">
      <Navbar.Brand>
        <img src={companyLogoPath} width="30px" height="30px" />
        Zinq Solutions
      </Navbar.Brand>
      <Navbar.Toggle aria-controls="responsive-navbar-nav" />
      <Navbar.Collapse id="responsive-navbar-nav">
        <Nav className="mr-auto">
          <LinkContainer to="/">
            <Nav.Link>Home</Nav.Link>
          </LinkContainer>
          <LinkContainer to="/not-found">
            <Nav.Link>Get Started</Nav.Link>
          </LinkContainer>
          <NavDropdown title="Dropdown" id="collasible-nav-dropdown">
            <NavDropdown.Item href="#action/3.1">Action</NavDropdown.Item>
            <NavDropdown.Item href="#action/3.2">
              Another action
            </NavDropdown.Item>
            <NavDropdown.Item href="#action/3.3">Something</NavDropdown.Item>
            <NavDropdown.Divider />
            <NavDropdown.Item href="#action/3.4">
              Separated link
            </NavDropdown.Item>
          </NavDropdown>
        </Nav>
        <Nav>
          <LinkContainer to="/register">
            <Nav.Link>Register</Nav.Link>
          </LinkContainer>
          <LinkContainer to="/login">
            <Nav.Link>Login</Nav.Link>
          </LinkContainer>
        </Nav>
      </Navbar.Collapse>
    </Navbar>
  );
};

export default NavigationBar;
