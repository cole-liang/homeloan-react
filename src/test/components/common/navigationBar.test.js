import React from "react";
import { shallow } from "enzyme";
import { NavigationBar } from "../../../components/common/navigationBar";
import testUsers from "../../fixture/users";

describe("NavigationBar Component Test", () => {
  it("should render NavigationBar component WITH user logged in", () => {
    const testUser = testUsers[0];
    const props = { user: testUser };
    const wrapper = shallow(<NavigationBar {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render NavigationBar component WITHOUT user logged in", () => {
    const props = { user: null };
    const wrapper = shallow(<NavigationBar {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render NavigationBar component WITH user login error", () => {
    const error = "Test error";
    const props = { user: { error } };
    const wrapper = shallow(<NavigationBar {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
