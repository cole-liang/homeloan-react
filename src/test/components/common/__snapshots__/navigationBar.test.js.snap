// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar Component Test should render NavigationBar component WITH user logged in 1`] = `
<styled.div>
  <Bootstrap(undefined)
    bg="dark"
    collapseOnSelect={true}
    expand="md"
    variant="dark"
  >
    <withRouter(LinkContainer)
      to="/"
    >
      <NavbarBrand>
        <img
          alt="Zinq Logo"
          src="http://www.zinqsolutions.com.au/wp-content/uploads/2018/12/zinq-comingsoon-logo2.png"
        />
        Zinq Solutions
      </NavbarBrand>
    </withRouter(LinkContainer)>
    <NavbarToggle
      aria-controls="responsive-navbar-nav"
      as="button"
      label="Toggle navigation"
    />
    <NavbarCollapse
      id="responsive-navbar-nav"
    >
      <Nav
        as="div"
        className="mr-auto"
        fill={false}
        justify={false}
      >
        <withRouter(LinkContainer)
          to="/getStarted/1"
        >
          <NavLink
            as={[Function]}
            disabled={false}
          >
            Get Started
          </NavLink>
        </withRouter(LinkContainer)>
      </Nav>
      <Nav
        as="div"
        fill={false}
        justify={false}
      >
        <div
          className="navDropdown mr-md-2"
        >
          <NavDropdown
            id="basic-nav-dropdown"
            title={
              <React.Fragment>
                <span
                  className="d-none d-md-flex align-items-center"
                >
                  <div
                    className="avatar"
                  />
                  <span>
                    Cole
                  </span>
                </span>
                <span
                  className="d-block d-md-none"
                >
                  <span>
                    Cole Liang
                  </span>
                </span>
              </React.Fragment>
            }
          >
            <DropdownItem
              as={[Function]}
              disabled={false}
              href="/logout"
            >
              Logout
            </DropdownItem>
          </NavDropdown>
        </div>
      </Nav>
    </NavbarCollapse>
  </Bootstrap(undefined)>
</styled.div>
`;

exports[`NavigationBar Component Test should render NavigationBar component WITH user login error 1`] = `
<styled.div>
  <Bootstrap(undefined)
    bg="dark"
    collapseOnSelect={true}
    expand="md"
    variant="dark"
  >
    <withRouter(LinkContainer)
      to="/"
    >
      <NavbarBrand>
        <img
          alt="Zinq Logo"
          src="http://www.zinqsolutions.com.au/wp-content/uploads/2018/12/zinq-comingsoon-logo2.png"
        />
        Zinq Solutions
      </NavbarBrand>
    </withRouter(LinkContainer)>
    <NavbarToggle
      aria-controls="responsive-navbar-nav"
      as="button"
      label="Toggle navigation"
    />
    <NavbarCollapse
      id="responsive-navbar-nav"
    >
      <Nav
        as="div"
        className="mr-auto"
        fill={false}
        justify={false}
      >
        <withRouter(LinkContainer)
          to="/getStarted/1"
        >
          <NavLink
            as={[Function]}
            disabled={false}
          >
            Get Started
          </NavLink>
        </withRouter(LinkContainer)>
      </Nav>
      <Nav
        as="div"
        fill={false}
        justify={false}
      >
        <withRouter(LinkContainer)
          to="/register"
        >
          <NavLink
            as={[Function]}
            disabled={false}
          >
            Register
          </NavLink>
        </withRouter(LinkContainer)>
        <withRouter(LinkContainer)
          to="/login"
        >
          <NavLink
            as={[Function]}
            disabled={false}
          >
            Login
          </NavLink>
        </withRouter(LinkContainer)>
      </Nav>
    </NavbarCollapse>
  </Bootstrap(undefined)>
</styled.div>
`;

exports[`NavigationBar Component Test should render NavigationBar component WITHOUT user logged in 1`] = `
<styled.div>
  <Bootstrap(undefined)
    bg="dark"
    collapseOnSelect={true}
    expand="md"
    variant="dark"
  >
    <withRouter(LinkContainer)
      to="/"
    >
      <NavbarBrand>
        <img
          alt="Zinq Logo"
          src="http://www.zinqsolutions.com.au/wp-content/uploads/2018/12/zinq-comingsoon-logo2.png"
        />
        Zinq Solutions
      </NavbarBrand>
    </withRouter(LinkContainer)>
    <NavbarToggle
      aria-controls="responsive-navbar-nav"
      as="button"
      label="Toggle navigation"
    />
    <NavbarCollapse
      id="responsive-navbar-nav"
    >
      <Nav
        as="div"
        className="mr-auto"
        fill={false}
        justify={false}
      >
        <withRouter(LinkContainer)
          to="/getStarted/1"
        >
          <NavLink
            as={[Function]}
            disabled={false}
          >
            Get Started
          </NavLink>
        </withRouter(LinkContainer)>
      </Nav>
      <Nav
        as="div"
        fill={false}
        justify={false}
      >
        <withRouter(LinkContainer)
          to="/register"
        >
          <NavLink
            as={[Function]}
            disabled={false}
          >
            Register
          </NavLink>
        </withRouter(LinkContainer)>
        <withRouter(LinkContainer)
          to="/login"
        >
          <NavLink
            as={[Function]}
            disabled={false}
          >
            Login
          </NavLink>
        </withRouter(LinkContainer)>
      </Nav>
    </NavbarCollapse>
  </Bootstrap(undefined)>
</styled.div>
`;
